corehq/sql_db/shard_data_management.py

```diff
@@ -99,6 +99,43 @@ def get_count_of_unmatched_models_by_shard(database, model):
     return results
 
 
+def delete_unmatched_shard_data(database, model):
```

Comment:
Any way to test this? It's a scary operation.

Reply:
https://github.com/dimagi/commcare-hq/pull/24763/commits/f06feb464a2edf279552cb895679abc848d741b6